### PR TITLE
Create QueryModel to fetch users of RDS with roles of member

### PR DESCRIPTION
**Pull request by harshith-venkatesh**

### CODE REVIEW

**Comment by harshith-venkatesh** [comment]
closes #384

**Comment by harshith-venkatesh** [comment]
closes #341

File: models/members.js [hunk 1]
```diff
@@ -98,8 +98,37 @@ const deleteIsMemberProperty = async () => {
   }
 }
 
+/**
+ * Fetches the data about our members with role member:true
+ * @return {Promise<userModel|Array>}
+ */
+
+const fetchOnlyMembers = async () => {
```
Comment (ankurnarkhede) [inline models/members.js:106]: How is this function different from the `fetchMembers` function defined already?
Can we reuse it?

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: models/members.js (hunk 1 above)
Comment (harshith-venkatesh) [reply models/members.js:106]: @ankurnarkhede , with the function `fetchMembers` we would be receiving the details of all the users along with their roles defined.
Upon scaling, we would rather be doing filter operation for millions of data which could be handled with db call while fetching required details right thus reducing computational time.
Open to suggestion

**Comment by ankurnarkhede** [comment]
In that case, the names of the functions needs to be changed. If the
function fetchMembers meant to fetch all users, not sure why the name is
fetchMembers and not fetchUsers. 🤔 Can you check if the functions using
the fetchMembers function actually need all users or just members, and
modify the function names accordingly? There doesn't seem any difference in
fetchMembers and fetchOnlyMembers for the reader 😅.

File: models/members.js (hunk 1 above)
Comment (harshith-venkatesh) [reply models/members.js:106]: We do have `fetchUsers` in users model due to which I went with different way of naming convention...
Shall I proceed with changing naming convention of `fetchMembers` to `fetchMembersWithAllRoles` and `fetchOnlyMembers` to `fetchMembersWithMemberRole` ?

Comment (sumitd94) [reply models/members.js:106]: just go with `fetchUsers`

**Comment by ankurnarkhede** [comment]
If the functions consuming them just require members, we can simply have
one function of fetchMembers and use it elsewhere.
Else, you can better create a function like fetchUsersWithRole(role=<role>)
and use it across the app.

**Comment by harshith-venkatesh** [comment]
@ankurnarkhede @sumitd94 changes are updated please review

File: constants/users.js [hunk 2]
```diff
@@ -1,2 +1,8 @@
 const userStatusEnum = ['ooo', 'idle', 'active']
-module.exports = userStatusEnum
+const ROLES = {
+  APPOWNER: 'app_owner',
+  DEFAULT: 'default',
+  MEMBER: 'member',
+  SUPERUSER: 'super_user'
+}
```
Comment (swarajpure) [inline constants/users.js:7]: Are we going to have `admin` role?

Comment (harshith-venkatesh) [reply constants/users.js:7]: These were the available roles that I had added
If we have documented about having admin role
I will add it accordingly @swarajpure

**Review by sumitd94 — DISMISSED** [review]
Looks good to me!

File: controllers/members.js [hunk 3]
```diff
@@ -10,7 +11,7 @@ const tasks = require('../models/tasks')
 
 const getMembers = async (req, res) => {
   try {
-    const allMembers = await fetchMembers()
+    const allMembers = await fetchUsers()
```
Comment (bhk31) [inline controllers/members.js:14]: What do you think `allMembers` is better or `users` or `allUsers` ?

File: controllers/members.js [hunk 4]
```diff
@@ -31,9 +32,9 @@ const getMembers = async (req, res) => {
 
 const getIdleMembers = async (req, res) => {
   try {
-    const allMembers = await fetchMembers()
+    const onlyMembers = await fetchUsersWithRole(ROLES.MEMBER)
```
Comment (bhk31) [inline controllers/members.js:35]: `idleMembers` would be better nomenclature according to function name

File: controllers/members.js (hunk 3 above)
Comment (harshith-venkatesh) [reply controllers/members.js:14]: I would go with `allMembers` here as we are actually assigning isMember value true or false property to it.

Comment (bhk31) [reply controllers/members.js:14]: We won't need to change API contracts and UI, just need to change variable name 

   ```
const allUsers = await fetchUsers()
     return res.json({
      message: allUsers.length ? 'Members returned successfully!' : 'No member found',
      members: allUsers
    })
```

Comment (harshith-venkatesh) [reply controllers/members.js:14]: I did understand the approach you had meant, I was not convinced enough with the change in variable names from allMembers to allUsers so far.

Comment (harshith-venkatesh) [reply controllers/members.js:14]: @bhk31 , I might be wrong here and would definitely stand to be corrected

Comment (bhk31) [reply controllers/members.js:14]: @harshith-venkatesh We can go ahead with either of the way. I don't have any problem with your approach. We can have opinion of @swarajpure @sumitd94 @ankurnarkhede  

@swarajpure @sumitd94 @ankurnarkhede Let us know your opinion.

Comment (harshith-venkatesh) [reply controllers/members.js:14]: It is actually a simple PR that is getting extended by a long time leading to prod issue of null data being observed.
I would rather make the changes to get this PR resolved as I am feeling bad looking at the idle member page being in that state

**Review by sumitd94 — APPROVED** [review]

**Pull request merged**